### PR TITLE
Firebase upload refactoring practice

### PR DESCRIPTION
**Pull request by reichert621**

### CODE REVIEW

File: src/kareem/core.clj [hunk 1]
```diff
@@ -26,38 +26,61 @@
...
+
+(defn get-acls []
+  (doto (ArrayList.) (.add (get-roles))))
+
+(defn build-blob-info [filename]
+  (-> (BlobInfo/newBuilder default-bucket filename)
+      (.setAcl (get-acls))
+      .build))
+
+(defn uri->stream [uri]
```
Comment (reichert621) [inline src/kareem/core.clj:53]: is this an idiomatic way to name a function that converts a uri to a stream in clojure?

File: src/kareem/core.clj [hunk 2]
```diff
@@ -26,38 +26,61 @@
...
+        filename (get-attachment-filename attachment)
+        blob (upload-file! uri filename)
+        firebase-uri (.getMediaLink blob)]
+    (assoc-in attachment [:payload :firebase-uri] firebase-uri)))
+
+(defn update-attachments! [message]
+  (update-in
+    message
+    [:message :attachments]
+    (fn [attachments] (map update-attachment! attachments))))
```
Comment (reichert621) [inline src/kareem/core.clj:83]: I'm curious why it has to be written like this rather than just passing `(map update-attachment!)` here?

**Comment by reichert621** [comment]
(merging for now so I can test deploying to heroku, but feel free to comment on this PR @stopachka)

File: src/kareem/core.clj (hunk 2 above)
Comment (stopachka) [reply src/kareem/core.clj:83]: hm, as `(map update-attachement!)` returns a function, that could work. If it doesn't we could also do: `(partial map update-attachement!)`

initial reason was, I thought we may have needed to do one more thing, outside of map

File: src/kareem/core.clj [hunk 3]
```diff
@@ -26,38 +26,61 @@
 
 (def default-bucket "kareem-2fdc3.appspot.com")
 
+(def type->ext
```
Comment (stopachka) [inline src/kareem/core.clj:29]: nice!

File: src/kareem/core.clj (hunk 1 above)
Comment (stopachka) [reply src/kareem/core.clj:53]: works for me :)

**Pull request merged**